### PR TITLE
Use Phoenix Live Components

### PR DESCRIPTION
**Pull request by SimonLab**
see #69

### CODE REVIEW

File: test/live_view_todo_web/live/page_live_test.exs [hunk 1]
```diff
@@ -17,20 +17,17 @@ defmodule LiveViewTodoWeb.PageLiveTest do
...
+    assert view |> element("#item-#{item.id}") |> render_click()
   end
 
   test "delete an item", %{conn: conn} do
     {:ok, item} = Item.create_item(%{"text" => "Learn Elixir"})
     assert item.status == 0
 
     {:ok, view, _html} = live(conn, "/")
-    assert render_click(view, :delete, %{"id" => item.id}) =~ "Todo"
+    view |> element("#delete-item-#{item.id}") |> render_click()
```
Comment (nelsonic) [inline test/live_view_todo_web/live/page_live_test.exs:30]: did you mean to remove the `assert` here?

File: README.md [hunk 2]
```diff
@@ -1483,7 +1484,156 @@ Borrow from:
 https://github.com/dwyl/phoenix-todo-list-tutorial#10-clear-completed
 
 
-## 11. Deploy to Heroku
+## 11. Live Components
+
+LiveView provides the Live Components feature to group UI state and events.
+In this section we're going to see how to use component for items.
+
+The first step if to create a new file `lib/live_view_todo_web/live/item_component.ex`:
```
Comment (nelsonic) [inline README.md:1492]:
```suggestion
The first step is to create a new file: 
`lib/live_view_todo_web/live/item_component.ex`
```

File: README.md [hunk 3]
```diff
@@ -1483,7 +1484,156 @@ Borrow from:
...
+
+```elixir
+defmodule LiveViewTodoWeb.ItemComponent do
+  use LiveViewTodoWeb, :live_component
+  alias LiveViewTodo.Item
+
+  attr(:items, :list, default: [])
+
+  def render(assigns) do
```
Comment (nelsonic) [inline README.md:1501]: _Reeeaaaallly_ not a fan of this `~H"""` templates inside the `render` function ... 😕 
is this the _only_ way of doing LiveComponents ...? 🤷‍♂️ 
This just feels like they've borrowed the _worst_ parts (from a maintainability perspective)
from React-land ...
can we not render partials and keep everything in `.heex` template files? 💭

File: test/live_view_todo_web/live/page_live_test.exs (hunk 1 above)
Comment (SimonLab) [reply test/live_view_todo_web/live/page_live_test.exs:30]: I can add back the assert :+1:

File: README.md (hunk 3 above)
Comment (SimonLab) [reply README.md:1504]: Indeed Phoenix has been inspired a lot by React, https://fly.io/blog/love-letter-react/

I've started by using [colocating templates](https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.html#module-colocating-templates), however I think the `attr` and `slot` (see https://hexdocs.pm/phoenix_live_view/Phoenix.Component.html#module-attributes) only work with using the `~H` sigil

Comment (nelsonic) [reply README.md:1504]: Hmm ... yes, I'd forgotten about that blog post ... 🙄 

```elixir
defmodule DemoWeb.CounterLive do
  use Phoenix.LiveView

  def render(assigns) do
    ~H"""
    <div>
      <p>You clicked <%= @count %> times</p>
      <button phx-click="inc">Click me</button>
    </div>
    """
  end

  def mount(_, _, socket), do: {:ok, assign(socket, count: 0)}

  def handle_event("inc"), do: {:noreply, update(socket, :count, &(&1 + 1))}
end
```
There's so much "magic" going on in this one file it's _suuuuuper_ unfriendly to beginners. 😢

File: test/live_view_todo_web/live/page_live_test.exs (hunk 1 above)
Comment (nelsonic) [reply test/live_view_todo_web/live/page_live_test.exs]: If it's not needed, fair enough. but if we aren't _assigning_ the result of this sequence to anything, 
then it's _unclear_ why it's there. 🤷‍♂️ 
So it _either_ needs a comment to the effect of:
```elixir
# Render the initial view then click on the item by id to delete it:
```
***or*** leave the `assert`. 😉

**Review by nelsonic — APPROVED** [review]
@SimonLab looks good. thanks for making the updates. 👌

**Pull request merged**